src/test/java/com/example/gildedrose/controller/ItemControllerTest.java

```diff
@@ -28,30 +30,47 @@
 @WebMvcTest(ItemController.class)
 class ItemControllerTest {
     @Autowired private MockMvc mockMvc;
+    @Autowired private Environment environment;
     @MockBean private ItemRepository itemRepository;
     @MockBean private Validators validators;
     private ObjectMapper mapper = new ObjectMapper();
 
     @Test
-    public void getsAllItems() throws Exception {
+    public void getsAllItemsWithAuthorizedUser() throws Exception {
         when(itemRepository.findAll()).thenReturn(new ArrayList<Updateable>(Arrays.asList(new NormalTestItem("Test"))));
+        this.mockMvc.perform(get("/items").with(httpBasic(environment.getProperty("USERNAME"), environment.getProperty("PASSWORD")))).andDo(print())
+            .andExpect(status().isOk())
+            .andExpect(content().string(containsString("Test")));
+    }
+
+    @Test
+    public void returnsUnauthorizedStatusWithUnauthorizedUserForGetRequest() throws Exception {
         this.mockMvc.perform(get("/items")).andDo(print())
-                .andExpect(status().isOk())
-                .andExpect(content().string(containsString("Test")));
+            .andExpect(status().isUnauthorized());
     }
 
     @Test
-    public void addsValidItem() throws Exception {
+    public void addsValidItemWithAuthorizedUser() throws Exception {
         NormalTestItem item = new NormalTestItem("Normal");
         ProposedItem proposedItem = new ProposedItem("Normal", "Normal", 0, 0);
         byte[] itemJson = mapper.writeValueAsBytes(proposedItem);
 
         when(validators.validate(Mockito.any(ProposedItem.class))).thenReturn(item);
         when(itemRepository.save(Mockito.any(Updateable.class))).thenReturn(item);
+        this.mockMvc.perform(post("/items").content(itemJson).contentType(MediaType.APPLICATION_JSON).accept(MediaType.APPLICATION_JSON).with(httpBasic(environment.getProperty("USERNAME"), environment.getProperty("PASSWORD"))))
+            .andDo(print())
```

Comment:
**suggestion**: Leave out the `andDo(print()))` to reduce noise in your test output.

I tend to use `andDo(print())` as a debugging tool as needed, and remove them once I am done.

Reply:
Okay